tests/lib/cli-engine/config-array/ignore-pattern.js

```diff
@@ -52,74 +52,79 @@ describe("IgnorePattern", () => {
+                    ["foo/bar/a.ts", false],
+                    ["foo/bar/b.js", false],
+                    ["foo/bar/b.ts", false],
+                    ["foo/bar/c.js", true],
+                    ["foo/bar/c.ts", true],
+                    ["foo/bar/dir/a.js", false],
+                    ["foo/bar/dir/a.ts", false],
+                    ["foo/bar/dir/b.js", true],
+                    ["foo/bar/dir/b.ts", false],
+                    ["foo/bar/dir/c.js", true],
+                    ["foo/bar/dir/c.ts", false],
+                    ["abc/a.js", false],
+                    ["abc/a.ts", false],
+                    ["abc/b.js", false],
+                    ["abc/b.ts", false],
+                    ["abc/c.js", true],
+                    ["abc/c.ts", true],
+                    ["abc/dir/a.js", false],
+                    ["abc/dir/a.ts", false],
+                    ["abc/dir/b.js", true],
+                    ["abc/dir/b.ts", false],
+                    ["abc/dir/c.js", true],
+                    ["abc/dir/c.ts", false]
+                ];
+
+                for (const [filename, expected] of patterns) {
+                    it(`should return ${expected} if '${filename}' was given.`, () => {
+                        assert.strictEqual(ignores(path.join(cwd, filename)), expected);
+                    });
+                }
+
+                it("should return false if '.dot.js' and false was given.", () => {
+                    assert.strictEqual(ignores(path.join(cwd, ".dot.js"), false), true);
                 });
-            }
 
-            it("should return false if '.dot.js' and false was given.", () => {
-                assert.strictEqual(ignores(path.join(cwd, ".dot.js"), false), true);
-            });
-
-            it("should return true if '.dot.js' and true were given.", () => {
-                assert.strictEqual(ignores(path.join(cwd, ".dot.js"), true), false);
-            });
+                it("should return true if '.dot.js' and true were given.", () => {
+                    assert.strictEqual(ignores(path.join(cwd, ".dot.js"), true), false);
+                });
 
+                it("should return false if '.dot/foo.js' and false was given.", () => {
+                    assert.strictEqual(ignores(path.join(cwd, ".dot/foo.js"), false), true);
+                });
 
-            it("should return false if '.dot/foo.js' and false was given.", () => {
-                assert.strictEqual(ignores(path.join(cwd, ".dot/foo.js"), false), true);
-            });
+                it("should return true if '.dot/foo.js' and true were given.", () => {
+                    assert.strictEqual(ignores(path.join(cwd, ".dot/foo.js"), true), false);
+                });
+            };
 
-            it("should return true if '.dot/foo.js' and true were given.", () => {
-                assert.strictEqual(ignores(path.join(cwd, ".dot/foo.js"), true), false);
-            });
+            assertions(process.cwd());
```

Comment:
I think we should have an explicit test for the Windows-style path, either by mocking `process.cwd()` or passing in a hardcoded string.

Reply:
line 127 - I went with `path.parse(process.cwd()).root` which will return the Drive on your windows CI nodes and reproduces failure when ignore-pattern.js does not contain the fix.

Reply:
let me know your thoughts on this or if you need it all mocked out.

Reply:
I tend to lean towards actually mocking it so that the tests are consistent regardless of the environment they’re run in, but if others disagree I’m happy to defer to consensus.

Reply:
I agree with Kai — it’s best to have the data a test uses be consistent regardless of the platform we run on.

Reply:
i think the problem with mocking so that you can always supply both "C:\\" on both linux and windows in this case is as is that it results in mimicking windows behavior of path.isAbsolute on windows when passing C: into path.isAbsolute is what was raising the issue to begin with.

this is as close as i can get on windows. can either Of you take it for a spin on a linux box to debug the issue?
```
        let didThrow = false;

            try {
                if (process.platform !== "win32") {
                    sinon.stub(process, "platform").returns("win32");
                    sinon.stub(path, "sep").returns("\\");
                    sinon.replace(path, "isAbsolute", pth => {
                        if (pth === "C:") {
                            throw Error("getCommonAncestor should never return C:, it should instead return C:\\");
                        } else {
                            return true;
                        }
                    });
                }
                assertions("C:\\");
            } catch {
                didThrow = true;
            } finally {
                sinon.restore();
            }
```

IMHO i think this type of mock is providing little value and the original ` assertions(path.parse(process.cwd()).root);` with no stubbing provides coverage of root C:\\ on windows but also additionally covers / on linux.

in any case if you can try take the snippet above onto a linux box and see if you can get it going as you desire that would be appreciated.

Reply:
@kaicataldo @nzakas - I have added an additional a test at the bottom of this file that stubs required path and process Apis for Linux and tested on Windows Subsystem Linux - Ubuntu 18.04. The new test will fail on both Windows and Linux locally if the fix to getCommonAncestors is removed.  All CI Gates/Checks are also passed. I think this is done, let me know otherwise.